terraform/modules/network/main.tf

```diff
@@ -0,0 +1,161 @@
+################################################################################
+# VPC Definition
+################################################################################
+resource "aws_vpc" "this" {
+  cidr_block                       = var.vpc_cidr
+  assign_generated_ipv6_cidr_block = true
+  enable_dns_support               = var.enable_dns_support
+  enable_dns_hostnames             = var.enable_dns_hostnames
+
+  tags = {
+    Name = "eda-vpc"
+  }
+}
+
+################################################################################
+# Availability zones
+################################################################################
+
+data "aws_availability_zones" "available" {
```

Comment:
hm this way we fetch all aws_availability_zones right? If we want to deploy something only to 2 of 3 availability_zones ?

Reply:
This is controlled by the variables `private_subnet_count` and `public_subnet_count` .